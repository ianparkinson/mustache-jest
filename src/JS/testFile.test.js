const renderToDiv = require('./testFile');

describe('renderToDiv', () => {
  let divElement;

  describe('when shouldShow is true', () => {
    beforeEach(() => {
      divElement = document.createElement('div');
      renderToDiv(divElement, { name: 'josie', shouldShow: true });
    });

    it('contains the content from the compiled template', () => {
      expect(divElement.innerHTML).toMatchSnapshot();
    });
  });

  describe('when shouldShow is false', () => {
    beforeEach(() => {
      divElement = document.createElement('div');
      renderToDiv(divElement, { name: 'josie', shouldShow: false });
    });

    it('contains the content from the compiled template', () => {
      expect(divElement.innerHTML).toMatchSnapshot();
    });
  });
});
