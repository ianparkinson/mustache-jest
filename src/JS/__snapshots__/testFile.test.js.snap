// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderToDiv when shouldShow is false contains the content from the compiled template 1`] = `
"<div>
  <p>Hello josie!</p>
  <div>GoodBye!!</div>
</div>
"
`;

exports[`renderToDiv when shouldShow is true contains the content from the compiled template 1`] = `
"<div>
  <p>Hello josie!</p>
  <div>Welcome!</div>
</div>
"
`;
